src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -817,14 +843,431 @@ public LsaPolicy()
+            SERVICE_START = 0x0010, // Required to call the StartService function to start the service. 
+            SERVICE_STOP = 0x0020, // Required to call the ControlService function to stop the service. 
+            SERVICE_PAUSE_CONTINUE = 0x0040, // Required to call the ControlService function to pause or continue the service. 
+            SERVICE_INTERROGATE = 0x0080, // Required to call the ControlService function to ask the service to report its status immediately. 
+            SERVICE_USER_DEFINED_CONTROL = 0x0100, // Required to call the ControlService function to specify a user-defined control code.
+
+            SERVICE_ALL_ACCESS = 0xF01FF, // Includes STANDARD_RIGHTS_REQUIRED in addition to all access rights in this table. 
+        }
+
+        [Flags]
+        public enum ServiceControlAccessRights : uint
+        {
+            SC_MANAGER_CONNECT = 0x0001, // Required to connect to the service control manager. 
+            SC_MANAGER_CREATE_SERVICE = 0x0002, // Required to call the CreateService function to create a service object and add it to the database. 
+            SC_MANAGER_ENUMERATE_SERVICE = 0x0004, // Required to call the EnumServicesStatusEx function to list the services that are in the database. 
+            SC_MANAGER_LOCK = 0x0008, // Required to call the LockServiceDatabase function to acquire a lock on the database. 
+            SC_MANAGER_QUERY_LOCK_STATUS = 0x0010, // Required to call the QueryServiceLockStatus function to retrieve the lock status information for the database.
+            SC_MANAGER_MODIFY_BOOT_CONFIG = 0x0020, // Required to call the NotifyBootConfigStatus function. 
+            SC_MANAGER_ALL_ACCESS = 0xF003F // Includes STANDARD_RIGHTS_REQUIRED, in addition to all access rights in this table. 
+        }            
+
+        public enum ServiceConfig2InfoLevel : uint
+        {
+            SERVICE_CONFIG_DESCRIPTION = 0x00000001, // The lpInfo parameter is a pointer to a SERVICE_DESCRIPTION structure.
+            SERVICE_CONFIG_FAILURE_ACTIONS = 0x00000002, // The lpInfo parameter is a pointer to a SERVICE_FAILURE_ACTIONS structure.
+            SERVICE_CONFIG_FAILURE_ACTIONS_FLAG = 0x00000004, // The lpInfo parameter is a pointer to a SERVICE_FAILURE_ACTIONS_FLAG structure.
+        }
+
+        public enum SC_ACTION_TYPE : uint
+        {
+            SC_ACTION_NONE = 0x00000000, // No action.
+            SC_ACTION_RESTART = 0x00000001, // Restart the service.
+            SC_ACTION_REBOOT = 0x00000002, // Reboot the computer.
+            SC_ACTION_RUN_COMMAND = 0x00000003 // Run a command.
+        }
+
+        [DllImport("advapi32.dll", CharSet = CharSet.Unicode)]
+        public static extern IntPtr OpenSCManager(string machineName, string db, ServiceControlAccessRights desiredAccess);
+
+        [DllImport("advapi32.dll", CharSet = CharSet.Unicode)]
+        public static extern IntPtr CreateService(
```

Comment:
Is this used? I haven't found any references. I'm not in favor of adding a bunch of methods/structs that aren't used. Please only add what you need.